akka-actor/src/main/scala/akka/io/TcpOutgoingConnection.scala

```diff
@@ -53,6 +54,7 @@ private[io] class TcpOutgoingConnection(
 
   def receive: Receive = {
     case registration: ChannelRegistration ⇒
+      setRegistration(registration)
```

Comment:
Necessary to keep our registration here early, so we can actually close the connection on error.